arbor/fvm_layout.hpp

```diff
@@ -87,7 +88,9 @@ struct fvm_discretization {
     }
 };
 
-fvm_discretization fvm_discretize(const std::vector<mc_cell>& cells);
+fvm_discretization fvm_discretize(
+        const std::vector<mc_cell>& cells,
+        const std::vector<fvm_index_type>& cell_to_intdom);
 
```

Comment:
The FVM discretization currently doesn't do anything for gap junctions (and probably won't ever have to), so the generation of the CV to intdom map from the `cell_to_intdom` vector and the discretization's cell to CV map is better off in the `fvm_lowered_cell_impl` code. This way we can keep the discretization code focused on the cell to CV mapping.

Reply:
Moved the code to fvm_lowered_cell_impl; I can no longer unit test cv_to_intdom so I removed the checks from the unit tests.